ui/perfherder/alerts/AlertsView.jsx

```diff
@@ -98,20 +122,48 @@ class AlertsView extends React.Component {
     return getStatus(parseInt(validated.status, 10));
   };
 
-  updateFramework = selection => {
-    const { updateParams } = this.props.validated;
-    const { frameworkOptions } = this.state;
-    const framework = frameworkOptions.find(item => item.name === selection);
-    updateParams({ framework: framework.id });
-    this.setState({ framework, bugTemplate: null }, () =>
-      this.fetchAlertSummaries(),
-    );
+  getDefaultFilterText = () => {
+    const { filterText } = this.props.validated;
+    return filterText === undefined || filterText === null ? '' : filterText;
   };
 
-  updateStatus = status => {
-    const statusId = summaryStatusMap[status];
-    this.props.validated.updateParams({ status: statusId });
-    this.setState({ status }, () => this.fetchAlertSummaries());
+  setParentState = filters => {
+    if (this.isListMode()) {
+      // also update params in url
+      this.props.validated.updateParams(this.getParamsFromFilters(filters));
+      this.setState(filters, this.fetchAlertSummaries);
+    } else {
+      this.setState(filters);
+    }
+  };
+
+  getParamsFromFilters = filters => {
+    const params = {};
+    for (const [filterName, filterValue] of Object.entries(filters)) {
+      Object.assign(params, this.mapParamFromFilter(filterName, filterValue));
+    }
+    return params;
+  };
+
+  // TODO: find a more elegant solution for this
+  mapParamFromFilter = (filterName, filterValue) => {
+    if (filterName === 'framework') {
```

Comment:
I think a basic `switch` statement would suffice, by improving readibility. Not sure it's worth adding a key-lambda dictionary.

Reply:
Also, a `switch` would replace the `if`s with `else if`s, thus making the check faster.